src/ert/_c_wrappers/config/config_parser.py

```diff
@@ -26,6 +52,43 @@ def __init__(self, errors: str, config_file: Optional[str] = None) -> None:
             else f"{self.errors}"
         )
 
+    @classmethod
+    def attempt(cls, *lambdas: [Callable[[], Any]]):
+        """
+        Helper function for running a set of functions and gathering up the errors.
+        To be used as an alternative to using a try-catch for every single
+        error-raising routine. It is assumed that every lambda function provided to
+        this is responsible for raising a ConfigValidationError (possibly combined)
+        when something goes wrong.
+        :param lambdas: the functions to be run
+        """
+        errors = []
+        results = []
+        for fn in lambdas:
+            try:
+                result = fn()
+                results.append(result)
+            except ConfigValidationError as err:
+                errors.append(err)
+                results.append(None)
+
+        return errors, results
+
+    @classmethod
+    def combined(cls, errors: ["ConfigValidationError"]):
+        """
+        Creates a combined ConfigValidationError with the same information as a list of
+        other ConfigValidationError instances.
+        :param errors a list of ConfigValidationErrors
+        """
```

Comment:
Instead of the `combined` method, I think we can use the separate `CombinedConfigError` I suggested above.